pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -308,3 +388,36 @@ func (s *AWSCloud) ensureLoadBalancerInstances(loadBalancerName string, lbInstan
 
 	return nil
 }
+
+func (s *AWSCloud) createProxyProtocolPolicy(loadBalancerName string) error {
+	glog.V(2).Info("XXX creating proxy protocol policy")
+
+	policyRequest := &elb.CreateLoadBalancerPolicyInput{
+		LoadBalancerName: aws.String(loadBalancerName),
+		PolicyName:       aws.String(ProxyProtocolPolicyName),
+		PolicyTypeName:   aws.String("ProxyProtocolPolicyType"),
+		PolicyAttributes: []*elb.PolicyAttribute{
+			{
+				AttributeName:  aws.String("ProxyProtocol"),
+				AttributeValue: aws.String("true"),
+			},
+		},
+	}
+	glog.V(2).Info("Creating proxy protocol policy on load balancer")
+	_, err := s.elb.CreateLoadBalancerPolicy(policyRequest)
+	if err != nil {
+		return fmt.Errorf("error creating proxy protocol policy on load balancer: %v", err)
+	}
+
+	return nil
+}
+
+func proxyProtocolEnabled(backend *elb.BackendServerDescription) bool {
+	for _, policy := range backend.PolicyNames {
+		if orEmpty(policy) == ProxyProtocolPolicyName {
```

Comment:
Let's use `aws.StringValue` instead of `orEmpty` for new code.  `orEmpty` is what I used when I didn't know about `aws.StringValue`.